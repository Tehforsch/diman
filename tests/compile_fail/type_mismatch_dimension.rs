#![allow(incomplete_features)]
#![feature(adt_const_params)]
#![feature(generic_const_exprs)]

use diman::{dimension, unit_system};

#[dimension]
pub struct Dimension {
    time: i32,
    length: i32,
}

#[rustfmt::skip]
unit_system!(
    Quantity,
    Dimension,
    [
        def Length = { length: 1 },
        unit (meters, "m") = Length,
        def Time = { time: 1 },

    ]
);

use crate::f64::*;

fn main() {
    let x: Time = Length::meters(1.0);
}
