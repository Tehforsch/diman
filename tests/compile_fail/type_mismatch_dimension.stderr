error[E0308]: mismatched types
 --> tests/compile_fail/type_mismatch_dimension.rs:7:19
  |
7 |     let x: Time = Length::meters(1.0);
  |                   ^^^^^^^^^^^^^^^^^^^ expected `example_system::Dimension { length: 0, time: 1, mass: 0, temperature: 0 }`, found `example_system::Dimension { length: 1, time: 0, mass: 0, temperature: 0 }`
  |
  = note: expected constant `example_system::Dimension { length: 0, time: 1, mass: 0, temperature: 0 }`
             found constant `example_system::Dimension { length: 1, time: 0, mass: 0, temperature: 0 }`
